tilavarauspalvelu/api/graphql/types/reservation/mutations.py

```diff
@@ -152,33 +154,47 @@ def validate_deletion(cls, reservation: Reservation, user: AnyUser) -> None:
             )
             raise ValidationError(msg)
 
-        payment_order: PaymentOrder = reservation.payment_order.first()
+        # Try Pindora delete first, if it fails, retry in background
+        try:
+            PindoraClient.delete_reservation(reservation=reservation)
```

Comment:
Should be after `validate_payment_order`: if that raises an error, the Pindora reservation is kept as well (until clean up by background task, if the reservation is not paid and confirmed in between).

Reply:
Fixed.